testing/gimmeproj/main.go

```diff
@@ -142,7 +145,21 @@ Administrative commands:
 		fmt.Fprintln(os.Stderr, usage.Error())
 		return nil
 	case "lease":
-		return lease(ctx, flag.Arg(1))
+		// When leasing, keep trying until we reach our configured timeout
+		ctx, cancel := context.WithTimeout(ctx, *waitTime)
+		defer cancel()
+		for ctx.Err() == nil {
+			err := lease(ctx, flag.Arg(1))
+			if err == nil {
+				return err
+			} else if errors.Is(err, ErrNoProjects) {
+				log.Printf("Temporary error: %v\n", err)
+				time.Sleep(30 * time.Second)
```

Comment:
future: Is 30 seconds the right delay? My experience is that Go testing currently takes longer than a minute. Thinking this through, this is probably a fairly cheap poll.

If we wanted to make it smarter, getting a retry-after direction from the lease function would be interesting, where we can add estimated test runtime to the oldest runner lease time.

Reply:
there's certainly room for improvement here, but i hesitate to add complexity to this tool directly. I think we'd be better served by building a server-side API component for this, which will also give us better visibility into client issues.

Reply:
Fair. I should have explained a bit more: I started with a real, optional feedback for right now but switched it to the "future" prefix when I realized I went too far, but still thought the notion worth sharing.